Assets/Src/Scripts/SeedCalc/GameManager.cs

```diff
@@ -14,23 +14,89 @@
 
 using System;
 using UnityEngine;
+using UnityEngine.UI;
 using AgileMvvm;
 
 namespace SeedCalc {
   public class GameManager : MonoBehaviour {
-    private Calculator _calculator = null;
-
-    // The UI component of the calculator screen must be associated to this field in Unity.
     public CalculatorScreen Screen;
     public CuttingBoard CuttingBoard;
+    public GameObject SettingsDialog;
+    public GameObject AboutDialog;
+    public Sprite ActiveButtonSprite;
+    public Sprite InactiveButtonSprite;
+
+    private Calculator _calculator = null;
+    private Button _chineseButton = null;
+    private Button _englishButton = null;
+    private Button _soundSwitchButton = null;
+    private GameObject _soundOnText = null;
+    private GameObject _soundOffText = null;
 
     public void OnClickButton(string input) {
       if (_calculator.AcceptingInput) {
-        GetComponent<AudioSource>().Play();
+        PlayClickSound();
         _calculator.OnInput(input);
       }
     }
 
+    public void OnOpenAbout() {
+      PlayClickSound();
+      AboutDialog.SetActive(true);
+      LocalizationUtils.UpdateLocalizedAssets(AboutDialog.gameObject, true);
+    }
+
+    public void OnCloseAbout() {
+      PlayClickSound();
+      AboutDialog.SetActive(false);
+    }
+
+    public void OnOpenSettings() {
+      PlayClickSound();
+      SettingsDialog.SetActive(true);
+      LocalizationUtils.UpdateLocalizedAssets(SettingsDialog.gameObject, true);
+      string langCode = LocalizationUtils.GetCurrentLocale();
+      if (langCode == LocalizationUtils.ChineseLangCode) {
+        SetButtonState(_chineseButton, true);
+        SetButtonState(_englishButton, false);
+      } else if (langCode == LocalizationUtils.EnglishLangCode) {
+        SetButtonState(_chineseButton, false);
+        SetButtonState(_englishButton, true);
+      } else {
+        Debug.Assert(false, $"{LocalizationUtils.GetCurrentLocale()}");
+      }
+      SetSoundSwitchState();
+    }
+
+    public void OnCloseSettings() {
+      PlayClickSound();
+      SettingsDialog.SetActive(false);
+    }
+
+    public void OnSetChinese() {
+      PlayClickSound();
+      if (LocalizationUtils.SetLocale(LocalizationUtils.ChineseLangCode)) {
+        SetButtonState(_chineseButton, true);
+        SetButtonState(_englishButton, false);
+      }
+    }
+
+    public void OnSetEnglish() {
+      PlayClickSound();
+      if (LocalizationUtils.SetLocale(LocalizationUtils.EnglishLangCode)) {
+        SetButtonState(_chineseButton, false);
+        SetButtonState(_englishButton, true);
+      }
+    }
+
+    public void OnSoundSwitch() {
+      AudioListener.volume = AudioListener.volume > 0.0f ? 0.0f : 1.0f;
```

Comment:
A strange expression (why 0 if the volume > 0, otherwise 1?):
```csharp
AudioListener.volume > 0.0f ? 0.0f : 1.0f
```

Reply:
Unity has no simple and straightforward switch to turn on/off the sound. It's just a volume adjustment. We don't want to provide a volume control UI. Instead, to provide a simple sound switch, it's a common way to: set the volume to zero if it's nonzero, and, set it to 1 if it's zero.